packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1543,6 +1545,11 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
       );
     }
 
+    result = PrimaryScrollController(
```

Comment:
You probably need more than just one controller per app, no? E.g. if you have multiple routes stacked on top of each other, they can all have primary scrollables, that need to scroll independent of each other?

Reply:
In that case, would this be better placed in the Navigator like it originally was in #69616?